react/week3/my-app/src/components/TodoList.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import Todo from "./Todo";
+import TodoInput from "./TodoInput";
+
+const TodoList = (props) => {
+  const data = props.todos.map((todo) => {
+    return { ...todo, isCompleted: false };
+  });
+
+  const [todos, setTodos] = useState(data);
+
+  const addTodo = (description, deadline) => {
+    const strDate =
+      deadline.getFullYear() +
+      "-" +
+      (deadline.getMonth() + 1) +
+      "-" +
+      deadline.getDate();
+
+    const newTodos = [...todos, { description, deadline: strDate }];
+    setTodos(newTodos);
+  };
+
+  const completeTodo = (index) => {
+    const newTodos = [...todos];
+    newTodos[index].isCompleted = !todos[index].isCompleted;
+    setTodos(newTodos);
+  };
+
+  const deleteTodo = (index) => {
+    const newTodos = [...todos];
+    newTodos.splice(index, 1);
+    setTodos(newTodos);
+  };
+  const updateTodo = (id, value) => {
+    const newTodo = todos.map((todo) => {
+      if (todo.id === id) {
+        return { ...todo, description: value };
+      }
+      return todo;
+    });
```

Comment:
UpdateTodo is not working properly. When we delete one task then click on another task update then you can find that todos list is not updated after deletion and it is taking old todos list.